lib/Runtime/Library/ArrayBuffer.cpp

```diff
@@ -1042,68 +1047,85 @@ namespace Js
         }
         uint32 growSize = newBufferLength - this->bufferLength;
 
-        bool failedReport = false;
-        const auto reportFailedFn = [&failedReport] { failedReport = true; };
+        struct AutoFailFastOnBadState
+        {
+            bool isStateValid = true;
+            ~AutoFailFastOnBadState()
```

Comment:
Ok, so you mean, in order to minimize the kind of things that can throw here, I should use `AutoDisableInterrupt` when it becomes a bad state so it won't failfast.
Do I get this right ?